nltk/tokenize/stanford_segmenter.py

```diff
@@ -75,8 +77,6 @@ def __init__(self,
 
         # This is passed to java as the -cp option, the old version of segmenter needs slf4j.
         # The new version of stanford-segmenter-2016-10-31 doesn't need slf4j
-        # When using old version, set path_to_slf4j='slf4j-api.jar' in the initilization of StanfordSegmenter
-        # The initilization program will search for your slf4j-api.jar according to the environment parameters you set
         self._stanford_jar = os.pathsep.join(
             [_ for _ in [stanford_segmenter, slf4j] if not _ is None])
```

Comment:
I'm not sure that a list needs to be built here, try
```python
self._stanford_jar = os.pathsep.join(
    _ for _ in [stanford_segmenter, slf4j] if _ is not None
)
```

Also `if not _ is None` is strange `_ is not None` is much better.